planet/src/main/webapp/js/placesItinerary.js

```diff
@@ -46,3 +59,71 @@ function createPlaceElement(ref, name, address) {
         </div>`;
     return placeElement;
 }
+
+function openAddPlaceForm(event, ref) {
+    const addPlaceForm = document.getElementById('add-place-to-event-form');
+
+    // Clone node to remove old event listeners
+    const addPlaceFormClone = addPlaceForm.cloneNode(true);
+    addPlaceForm.parentNode.replaceChild(addPlaceFormClone, addPlaceForm);
+
+    addPlaceFormClone.style.display = 'block';
+    addPlaceFormClone.style.left = event.clientX + 'px';
+    addPlaceFormClone.style.top = event.clientY + 'px';
+    document.getElementById('submit-place').addEventListener('click', () => {
+        submitPlace(ref);
+    });
+}
+
+function closeAddPlaceForm() {
+    const addPlaceForm = document.getElementById('add-place-to-event-form');
+    addPlaceForm.style.display = 'none';
+}
+
+async function submitPlace(ref) {
+    const eventDuration = document.getElementById('add-place-duration').value;
+
+    // Validate the input duration
+    if (!validatePlaceDuration(eventDuration)){
+        return;
+    }
+
+    const userId = firebase.auth().currentUser.uid;
+    const listName = document.getElementById('list-options').value;
+    const eventListRef = database.ref('events/' + userId + '/' + listName);
+    // Get the order number by counting existing events
+    const eventListSnapshot = await eventListRef.once('value');
+    const order = eventListSnapshot.numChildren() + 1;
+
+    //Create a new event based on the place
+    const placeRef = database.ref('places/' + ref); 
+    const placeSnapshot = await placeRef.once('value');
+    const placeDetails = placeSnapshot.val();
+
+    const newEventRef = eventListRef.child(ref);
+    newEventRef.set({
+        name: placeDetails.name,
+        address: placeDetails.address,
+        duration: eventDuration,
+        openingTime: placeDetails.openingTime,
+        closingTime: placeDetails.closingTime,
+        order: order,
+    });
+    closeAddPlaceForm();
+    renderPlaces();
+}
+
+function validatePlaceDuration(duration) {
+    let isValid = true;
```

Comment:
it looks like this variable isn't needed. Anytime it's set to false, we immediately return. We can just directly return false or true instead.

Reply:
Fixed!